### PR TITLE
More unit test and add more linters

### PR DESCRIPTION
**Pull request by asalkeld**
part of https://github.com/nitrictech/nitric/issues/256

### CODE REVIEW

File: pkg/providers/aws/plugin.go [hunk 1]
```diff
@@ -25,26 +25,13 @@ import (
...
-
-	if err != nil {
-		return nil, err
-	}
-
-	return &AWSServiceFactory{
-		provider: provider,
-	}, nil
-}
-
```
Comment (tjholm) [inline pkg/providers/aws/plugin.go:47]: If the code in these is dead, we can probably remove these.

Initially wanted to look at using something like `wire` to do compile time dependency injection for building the different provider membranes, but we can clean out this dead code for now.

File: pkg/adapters/grpc/faas_grpc.go [hunk 2]
```diff
@@ -88,8 +88,10 @@ func (s *FaasServer) TriggerStream(stream pb.FaasService_TriggerStreamServer) er
 	log.Default().Println("FaaS stream closed, removing worker")
 
 	// Worker is done so we can remove it from the pool
-	s.pool.RemoveWorker(wrkr)
-
+	rwErr := s.pool.RemoveWorker(wrkr)
+	if err == nil {
+		err = rwErr
+	}
 	return err
```
Comment (tjholm) [inline pkg/adapters/grpc/faas_grpc.go:95]: good that error handling is being added here, but the logic seems a little confusing, if we experience an error calling pool.RemoveWorker, but we had another error before, we're justing throwing away the new error?

Comment (asalkeld) [reply pkg/adapters/grpc/faas_grpc.go:95]: yeah, I wasn't sure what the priority was.. of if they were dependent. I could have an error list here..

**Review by davemooreuws — APPROVED** [review]
Nice work Angus! Approved, Tims suggestions can go in here or new pr.

File: pkg/adapters/grpc/faas_grpc.go (hunk 2 above)
Comment (asalkeld) [reply pkg/adapters/grpc/faas_grpc.go:95]: improvement comming

File: pkg/plugins/storage/boltdb/storage.go [hunk 3]
```diff
@@ -166,21 +137,6 @@ func (s *BoltStorageService) Delete(bucket string, key string) error {
...
-		)
-	}
-	if key == "" {
-		return newErr(
-			codes.InvalidArgument,
-			"provide non-blank key",
-			nil,
-		)
-	}
-
```
[the remark this answers is not in the document]
Comment (asalkeld) [reply pkg/plugins/storage/boltdb/storage.go:183]: removed

**Review by tjholm — APPROVED** [review]

**Pull request merged**